docs/general_usage_patterns.md

```diff
@@ -0,0 +1,213 @@
+in which we should use them.
+
+
+## General structure
+
+Each regular module has four different sets of parameters that contain
+information about the ServiceNow instance we will manage, record identifiers,
+commonly-used resource fields, and a catch-all parameter for custom data.
+
+The first set contains a single parameter called `instance` that includes:
+
+ 1. Instance address, e.g. `https://dev12345.service-now.com/`.
+ 2. Username and password that Ansible modules should use to authenticate with
+    the ServiceNow instance.
+ 3. The optional client id and secret that indicate modules should use OAuth
+    when authenticating.
+
+Parameters in the second set allow us to select an existing record to manage.
+All modules support identifying records by system id (the `sys_id` parameter).
+Because those are often awkward to work with, most modules know how to select
+records by their number. Those numbers are readily available from the
+ServiceNow user interface.
+
+The third set of parameters offers convenient access to commonly-used record
+fields. Some of the parameters from the third set contain additional validation
+logic that helps guide playbook authors. For example, the `state` parameter
+only accepts valid state names.
+
+The `other` parameter forms the last set of parameters. This free-form
+parameter allows us to pass any data to the backend. This parameter comes in
+handy when we want to set data in user-created custom columns.
+
+Info modules are more spartan and only have two sets of parameters: an
+`instance` parameter and record selectors like `sys_id` and `number`.
+
+In practice, tasks using a module from ServiceNow ITSM Ansible Collection looks
+like this:
+
+    ---
+    - name: Sample playbook
+      hosts: localhost
+      gather_facts: false
+
+      tasks:
+	- name: Create a new incident
+	  servicenow.itsm.incident:
+            # Instance data
+            instance:
+              host: https://dev12345.service-now.com
+              username: user
+              password: pass
+              client_id: cid
+              client_secret: csecret
+
+            # No sys_id or number because we are creating a new incident
+
+            # Common parameters
+	    caller: admin
+	    short_description: Demo incident
+	    impact: low
+	    urgency: low
+
+            # Other data
+            other:
+              problem_id: "{{ problem.record.sys_id }}"
+
+	- name: Update an existing incident
+	  servicenow.itsm.incident:
+            # Instance data
+            instance:
+              host: https://dev12345.service-now.com
+              username: user
+              password: pass
+              client_id: cid
+              client_secret: csecret
+
+            # Record identificator
+            number: INC1234567
+
+            # Common parameters
+            state: in_progress
+	    impact: high
+	    urgency: high
+
+	- name: Delete incident ticket (rare operation)
```

Comment:
Can you please add these playbooks as separate files and include them in the documentation like https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/scenario_guides/vmware_rest_scenarios/create_vm.rst

Reply:
Unfortunately, I cannot do this in markdown documents. It would be great if Automation Hub could render rST documentation, but currently, the documentation display is limited to markdown.